src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +252,15 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+  private void flushMessageBuffer() {
+    int batchSize = Math.max(1 , config.getBatchSize());
```

Comment:
where are we having handling for other message formats, delimeters  also need to be handled for others and for JSON type, comma will be only delimiter.

Reply:
JSON and Form format have well defined structure and parsing library depend on it.
having a custom delimiters only makes sense with custom message type where the user will be parsing the message themselves